src/VisualStudio/Core/Def/Storage/VisualStudioCloudCacheServiceProvider.cs

```diff
@@ -0,0 +1,49 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Composition;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.Editor.Shared.Utilities;
+using Microsoft.CodeAnalysis.Host.Mef;
+using Microsoft.CodeAnalysis.Storage;
+using Microsoft.VisualStudio.RpcContracts.Caching;
+using Microsoft.VisualStudio.Shell;
+using Microsoft.VisualStudio.Shell.ServiceBroker;
+using Roslyn.Utilities;
+
+namespace Microsoft.VisualStudio.LanguageServices.Storage
+{
+    [ExportWorkspaceService(typeof(ICloudCacheServiceProvider), ServiceLayer.Host), Shared]
+    internal class VisualStudioCloudCacheServiceProvider : ICloudCacheServiceProvider
+    {
+        private readonly IAsyncServiceProvider _serviceProvider;
+        private readonly IThreadingContext _threadingContext;
+
+        [ImportingConstructor]
+        [Obsolete(MefConstruction.ImportingConstructorMessage, error: true)]
+        public VisualStudioCloudCacheServiceProvider(
+            IThreadingContext threadingContext,
+            SVsServiceProvider serviceProvider)
+        {
+            _threadingContext = threadingContext;
+            _serviceProvider = (IAsyncServiceProvider)serviceProvider;
+        }
+
+        public async ValueTask<ICloudCacheService> CreateCacheAsync(CancellationToken cancellationToken)
+        {
+            var serviceContainer = await _serviceProvider.GetServiceAsync<SVsBrokeredServiceContainer, IBrokeredServiceContainer>().ConfigureAwait(false);
+            var serviceBroker = serviceContainer.GetFullAccessServiceBroker();
+
+#pragma warning disable ISB001 // Dispose of proxies
+            // cache service will be disposed inside VisualStudioCloudCacheService.Dispose
+            var cacheService = await serviceBroker.GetProxyAsync<ICacheService>(VisualStudioServices.VS2019_10.CacheService, cancellationToken: cancellationToken).ConfigureAwait(false);
```

Comment:
❔ Have you verified that _everything_ above this uses JTF and does not try to use `WaitAndGetResult`?

Reply:
no.  i'm not 100% certain.